### PR TITLE
Fix validation in block selection and implement more tests

### PR DESCRIPTION
**Pull request by akhila-raju**
Introducing new tests:

- error when selecting block with no number, hash or tag.
- error when selecting block with more than one argument (number, hash, tag).
- error when selecting blocks with both numbers and hashes.
- error when selecting blocks with no numbers or hashes.
- error when selecting blockRange with starting or ending block being negative numbers.
- account->balance selection.

[ch158]

### CODE REVIEW

File: src/__tests__/core/account.test.ts [hunk 1]
```diff
@@ -13,7 +13,6 @@ test('account: select by address', async () => {
   `;
 
   const expected = { data: { account: { code: '0x', transactionCount: 0 } } };
-
```
Comment (raulk) [inline src/__tests__/core/account.test.ts]: This is an unnecessary change?

File: src/__tests__/core/account.test.ts [hunk 2]
```diff
@@ -32,3 +31,17 @@ test('account: error when address is invalid', async () => {
...
+    {
+      account(address: "0x0000000000000000000000000000000000000000") {
+        balance
+      }
+    }
+  `;
+
+  const result = await execQuery(query);
+  expect(result.errors).toBeUndefined();
+  expect(result.data.account.balance).toBeGreaterThan(0);
```
Comment (raulk) [inline src/__tests__/core/account.test.ts]: Cool, this test is more solid ;-)

File: src/model/EthqlQuery.ts [hunk 3]
```diff
@@ -36,7 +36,8 @@ class EthqlQuery {
     tag = tag ? tag.trim().toLowerCase() : tag;
 
     const params = _.reject([blockNumber, hash, tag], _.isNil);
-    if (!args) {
+
+    if (!params.length) {
```
Comment (raulk) [inline src/model/EthqlQuery.ts]: Nice catch here!

Comment (akhila-raju) [reply src/model/EthqlQuery.ts]: Thanks :D

File: src/__tests__/core/account.test.ts (hunk 2 above)
Comment (akhila-raju) [reply src/__tests__/core/account.test.ts]: Woo!
